### PR TITLE
Worked on the typos

### PR DESCRIPTION
**Pull request by srpcoder**
Changed all the "MB" --> "MiB", "KB" --> "KiB" , "GB" -->  "GiB"

### CODE REVIEW

File: src/flags.zig [hunk 1]
```diff
@@ -307,10 +307,10 @@ fn parse_value_size(flag: []const u8, value: []const u8) ByteSize {
...
 
     const units = .{
-        .{ &[_][]const u8{ "TiB", "tib", "TB", "tb" }, 1024 * 1024 * 1024 * 1024 },
-        .{ &[_][]const u8{ "GiB", "gib", "GB", "gb" }, 1024 * 1024 * 1024 },
-        .{ &[_][]const u8{ "MiB", "mib", "MB", "mb" }, 1024 * 1024 },
-        .{ &[_][]const u8{ "KiB", "kib", "KB", "kb" }, 1024 },
+        .{ &[_][]const u8{ "TiB", "tib", "TiB", "tb" }, 1024 * 1024 * 1024 * 1024 },
+        .{ &[_][]const u8{ "GiB", "gib", "GiB", "gb" }, 1024 * 1024 * 1024 },
+        .{ &[_][]const u8{ "MiB", "mib", "MiB", "mb" }, 1024 * 1024 },
+        .{ &[_][]const u8{ "KiB", "kib", "KiB", "kb" }, 1024 },
```
Comment (matklad) [inline src/flags.zig]: * Remove the duplicate variants --- it's enough to have only one `TiB` here
* Remove lower-case two-letter variants like `tb` --- we want to only allow i-ones.

**Review by matklad — CHANGES_REQUESTED** [review]

File: src/flags.zig (hunk 1 above)
Comment (srpcoder) [reply src/flags.zig]: - Duplicates removed and the lower case values are changed.

**Comment by matklad** [comment]
Looks like not everything was updated, and that the tests should fail:

```
λ rg -e '--cache-grid=.*MB' .
./docs/quick-start/with-docker-compose.md
123:less memory. For example, set `--cache-grid=512MB` when running `tigerbeetle start`.

./src/tigerbeetle/cli.zig
82:        \\  tigerbeetle start --addresses=<addresses> [--cache-grid=<size><KB|MB|GB>] <path>
129:        \\  --cache-grid=<size><KB|MB|GB>

./src/testing/tmp_tigerbeetle.zig
80:        "{tigerbeetle} start --cache-grid=512MB --addresses=0 {data_file}",

./src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs
1594:        process.StartInfo.Arguments = $"start --addresses=0 --cache-grid=512MB ./{dataFile}";

./src/clients/go/tb_client_test.go
43:	cacheSizeArg := "--cache-grid=512MB"

./src/clients/java/src/test/java/com/tigerbeetle/IntegrationTest.java
1641:                    .command(new String[] {exe, "start", "--addresses=0", "--cache-grid=512MB",
```

Please carefully look for all other usages of old suffixes and update them if necessary. Try running `./zig/zig build scripts -- ci`  and ``./zig/zig build test` locally to make sure that the build passes.

**Comment by brson** [comment]
Just linking this to

- https://github.com/tigerbeetle/tigerbeetle/issues/1672
- https://github.com/tigerbeetle/tigerbeetle/issues/1673

as it looks to solve both.

**Comment by srpcoder** [comment]
Hi,
The changes are made. Kindly review.

**Comment by matklad** [comment]
It seems like there's at least

```
TIGERBEETLE_CACHE_GRID_SIZE=1GB
```

left in tigerbeetle.service

@srpcoder try grepping for `=.*GB` and all other variants to make sure all usages are updated. 

That's the hard bit here -- figuring out just _what_ needs to change. One tip there is to search things in some automated fashion, using grep or rg, and then add the commands used in the commit message. That way, the reviewer can be more certain that all required usages are properly updated.

**Comment by srpcoder** [comment]
Yeah let me try it out

File: src/lsm/manifest_log.zig [hunk 2]
```diff
@@ -1055,9 +1055,9 @@ pub const Options = struct {
 /// - After compacting `T+2` blocks, we must have freed at least 2 whole blocks.
 /// - Etc.
 ///
-/// Then the upper-bound number of manifest blocks (`MB(b)`) at any half-bar boundary (`b`) is:
+/// Then the upper-bound number of manifest blocks (`MiB(b)`) at any half-bar boundary (`b`) is:
 ///
-///   MB(b) = min(T, MB(b-1)) + A×⌈M(b-1)/C⌉ + A×⌈(T+1)/C⌉
```
Comment (matklad) [inline src/lsm/manifest_log.zig]: This one is wrong, MB here is "Manifest Blocks", not "mega bytes"

Comment (srpcoder) [reply src/lsm/manifest_log.zig]: Sorry @matklad updated in src/lsm/manifest_log.zig

**Comment by matklad** [comment]
I think only one more left:

```
src/flags.zig
964:    try t.check(&.{ "values", "--size=100000000000000000kb" }, snap(@src(),
967:        \\error: --size: size in bytes exceeds 64-bit unsigned integer: '100000000000000000kb'
```

```
λ ./zig/zig build test -- flags
run test: error: 'test.flags' failed: Snapshot differs.
Want:
----
status: 1
stderr:
error: --size: size in bytes exceeds 64-bit unsigned integer: '100000000000000000kb'

----
Got:
----
status: 1
stderr:
error: --size: expected a size, but found '100000000000000000kb' (invalid digit or suffix)

----
Rerun with SNAP_UPDATE=1 environmental variable to update the snapshot.
/home/matklad/p/tb/review/src/testing/snaptest.zig:158:13: 0x236bcf in diff (test)
            return error.SnapDiff;
            ^
/home/matklad/p/tb/review/src/flags.zig:647:13: 0x239348 in check (test)
            try want.diff(t.output_buf.items);
            ^
/home/matklad/p/tb/review/src/flags.zig:964:5: 0x23b06d in test.flags (test)
    try t.check(&.{ "values", "--size=100000000000000000kb" }, snap(@src(),
    ^
run test: error: while executing test 'test.flags', the following test command failed:
/home/matklad/p/tb/review/zig-cache/o/93f7c83fc6d084700970328a42d07b81/test --listen=- 
Build Summary: 4/6 steps succeeded; 1 failed; 0/1 tests passed; 1 failed (disable with --summary none)
test transitive failure
└─ run test 0/1 passed, 1 failed
error: the following build command failed with exit code 1:
/home/matklad/p/tb/review/zig-cache/o/c6e4f6e1d1ffc5e7f85f3c8981f51e09/build /home/matklad/p/tb/review/zig/zig /home/matklad/p/tb/review /home/matklad/p/tb/review/zig-cache /home/matklad/.cache/zig test -- flags
```

**Comment by matklad** [comment]
Tests still fail:

```
λ ./zig/zig build test
run test: error: error: --size: expected a size, but found '1000mb' (invalid digit or suffix)
run test: error: while executing test 'decltest.parse_value_size', the following command exited with code 1 (expected exited with code 0):
/home/matklad/p/tb/review/zig-cache/o/b8963df68820d85f368238d56b1c51ba/test --listen=- 
Build Summary: 17/19 steps succeeded; 1 failed; 195/195 tests passed (disable with --summary none)
test transitive failure
└─ run test failure
error: the following build command failed with exit code 1:
/home/matklad/p/tb/review/zig-cache/o/c6e4f6e1d1ffc5e7f85f3c8981f51e09/build /home/matklad/p/tb/review/zig/zig /home/matklad/p/tb/review /home/matklad/p/tb/review/zig-cache /home/matklad/.cache/zig test
```

**Review by matklad — APPROVED** [review]
Looks good now I think, thanks @srpcoder!

**Pull request merged**